mailet/api/src/main/java/org/apache/mailet/MailetConfig.java

```diff
@@ -40,7 +41,7 @@
      * @param name the name of the initialization parameter
      * @return the value of the initialization parameter, or null
      */
-    String getInitParameter(String name);
+    Object getInitParameter(String name);
```

Comment:
I tend to strongly believe strong typing is a nice property to have, as it enables compile time checks and improves safety + efficiency. It also improves API discovery, which is actually quite critical in an API that is the base of James extension model.

We should refrain from changing any API related to mailets.

That being said, maybe we can add a : 

```
<T> T getInitParameter(String name, Class<T> clazz) throws ClassCastException;
```

Might be a good idea, as it might allow the caller to avoid some job parsing strings.

Note that this method returns always the right type, which is now decided by the caller (and not by the underlying datatype).

Also, I would be in favor to keep the 'old' code using the previous 'String' version....

Reply:
Is the `Class<T> clazz` in the `getInitParameter` used to indicate the data type for the returned value to improve API discovery? If yes, what supposed to fill in the `config.getInitParameter(name, )` used in the method `getInitParameter` in class `GenericMailet`?

Reply:
The reason that I use the enum version is that I think using String might cause problems when fill string with different cases by mistake, but anyway, I'll roll the related code back.

Reply:
In `GenericMailet` it calls the` getInitParameter` in the `MailetConfig`, however, when added the parameter `Class<T> clazz`, there's something needed to be passed in or construct when calling the method, I don't know what to pass here.

Reply:
Well, this method needs to be implemented in MailetConfig & MailetConfigImpl, then GenericMailet only needs to be calling that:

```
    @Override
    public <T> T getInitParameter(String name, Class<T> clazz) throws ClassCastException {
        return getMailetConfig().getInitParameter(name, clazz);
    }
```

For the record, in MailetConfigImpl, this could look like this: 

```
    @Override
    public <T> T getInitParameter(String name, Class<T> clazz) throws ClassCastException {
        Object o = configuration.getProperty(name);
        if (clazz.isInstance(o)) {
            return (T) o;
        }
        throw new ClassCastException(name + " property is not of type " + clazz.getCanonicalName());
    }
```

Would this seems Ok to you?